ToDoList-Redux/Binders/TaskListViewModel.swift

```diff
@@ -65,25 +57,25 @@ extension Reducer where ActionType == TaskList.Action, StateType == TaskList.Sta
 // MARK: - PROJECTIONS
 
 extension CheckmarkCellView {
-    
     static func viewModel<S: StoreType>(store: S, taskId: String) -> ObservableViewModel<CheckmarkCellView.Action, CheckmarkCellView.State>
-    where S.ActionType == TaskList.Action, S.StateType == TaskList.State {
-        let task = store.mapState { state in state.tasks.first { $0.id == taskId } }
+    where S.ActionType == TaskAction, S.StateType == [Task] {
+        let task = store.mapState { state in state.first { $0.id == taskId } }
         return task
-            .projection(action: Self.transform, state: Self.transform)
+            .projection(action: Self.transform(taskId: taskId), state: Self.transform)
             .asObservableViewModel(initialState: .empty)
     }
     
-    static func transform(_ viewAction: CheckmarkCellView.Action) -> TaskList.Action? {
-        switch viewAction {
-            case let .toggle(id): return .toggle(.toggle(id))
+    static func transform(taskId: String) -> (CheckmarkCellView.Action) -> TaskAction? {
+        return { viewAction in
+            switch viewAction {
+            case .toggle: return .toggle(taskId)
```

Comment:
And suddenly your View Event doesn't need the id any more. We can create a AppAction.TaskAction (that needs the id) out of this View Event that doesn't hold the id. So the row no longer needs the id for anything.